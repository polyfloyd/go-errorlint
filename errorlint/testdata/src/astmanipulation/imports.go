package astmanipulation

import (
	"io"
	"os"
)

// Test missing errors import with simple comparison
func MissingImportSimple() {
	var err error
	var n int

	if n == 0 || (err != nil && err != io.EOF) { // want "comparing with != will fail on wrapped errors. Use errors.Is to check for a specific error"
		return
	}
}

// Test missing errors import with type assertion
func MissingImportTypeAssertion() {
	var err error

	if e, ok := err.(*os.PathError); ok { // want "type assertion on error will fail on wrapped errors. Use errors.As to check for specific errors"
		println("Path error:", e.Path)
	}
}