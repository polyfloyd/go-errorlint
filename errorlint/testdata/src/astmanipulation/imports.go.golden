package astmanipulation

import (
	"errors"
	"io"
	"os"
)

// Test missing errors import with simple comparison
func MissingImportSimple() {
	var err error
	var n int
	
	if n == 0 || (err != nil && !errors.Is(err, io.EOF)) { // want "comparing with != will fail on wrapped errors. Use errors.Is to check for a specific error"
		return
	}
}

// Test missing errors import with type assertion
func MissingImportTypeAssertion() {
	var err error
	
	e := &os.PathError{}
	if errors.As(err, &e) { // want "type assertion on error will fail on wrapped errors. Use errors.As to check for specific errors"
		println("Path error:", e.Path)
	}
}